model/pipeline/src/main/resources/org/apache/beam/model/pipeline/pipeline.json

```diff
@@ -0,0 +1,320 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+
+  "type": "object",
+
+  "$ref": "#/definitions/Pipeline",
+
+  "definitions": {
+    // A Pipeline contains the entire hierarchical graph including
+    // nodes for all values.
+    //
+    // Nodes (transform or value or other) are identified by some unique id. This need
+    // not be readable. It is simplest to think of it as a reified pointer or UUID.
+    // Cyclic pointer structures are forbidden throughout.
+    "Pipeline": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": [ "userFns", "transforms", "values", "windowingStrategies", "coders" ],
+      "properties": {
+
+        // A mapping containing all user-definable functions in the pipeline.
```

Comment:
Should all objects have a unique id across the different types (userFns, transforms, values, ...)?